// @wix/site-stores/dist/cjs/cart.runtime.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cartRuntime = void 0;
const host_modules_1 = require("@wix/sdk-runtime/host-modules");
exports.cartRuntime = (0, host_modules_1.createHostModule)({
    addProducts: ({ channel }) => function addProducts(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.addProducts',
            args,
        });
    },
    applyCoupon: ({ channel }) => function applyCoupon(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.applyCoupon',
            args,
        });
    },
    getCurrentCart: ({ channel }) => function getCurrentCart(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.getCurrentCart',
            args,
        });
    },
    hideMiniCart: ({ channel }) => function hideMiniCart(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.hideMiniCart',
            args,
        });
    },
    onChange: ({ channel }) => function onChange(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.onChange',
            args,
        });
    },
    removeCoupon: ({ channel }) => function removeCoupon(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.removeCoupon',
            args,
        });
    },
    removeProduct: ({ channel }) => function removeProduct(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.removeProduct',
            args,
        });
    },
    showMiniCart: ({ channel }) => function showMiniCart(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.showMiniCart',
            args,
        });
    },
    updateLineItemQuantity: ({ channel }) => function updateLineItemQuantity(...args) {
        return channel.invoke({
            namespace: 'stores',
            method: 'cart.updateLineItemQuantity',
            args,
        });
    },
});
