// @wix/site-stores/dist/cjs/cart.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cartContext = void 0;
const context_1 = require("@wix/sdk-runtime/context");
const cart_runtime_1 = require("./cart.runtime");
exports.cartContext = {
    addProducts: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'addProducts'),
    applyCoupon: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'applyCoupon'),
    getCurrentCart: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'getCurrentCart'),
    hideMiniCart: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'hideMiniCart'),
    onChange: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'onChange'),
    removeCoupon: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'removeCoupon'),
    removeProduct: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'removeProduct'),
    showMiniCart: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'showMiniCart'),
    updateLineItemQuantity: (0, context_1.contextualizeHostModule)(cart_runtime_1.cartRuntime, 'updateLineItemQuantity'),
};
