// @wix/auto_sdk_stores_catalog-versioning/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getCatalogVersion: () => getCatalogVersion2
});
module.exports = __toCommonJS(meta_exports);

// src/stores-catalog-v3-provision-catalog-versioning.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixStoresCatalogProvisionV1StoresProvisionServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      },
      {
        srcPath: "/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    _: [
      {
        srcPath: "/_api/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_catalog-versioning";
function getCatalogVersion(payload) {
  function __getCatalogVersion({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.provision",
      method: "GET",
      methodFqn: "wix.stores.catalog.provision.v1.StoresProvisionService.GetCatalogVersion",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogProvisionV1StoresProvisionServiceUrl({
        protoPath: "/v3/provision/version",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCatalogVersion;
}

// src/stores-catalog-v3-provision-catalog-versioning.meta.ts
function getCatalogVersion2() {
  const payload = {};
  const getRequestOptions = getCatalogVersion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/provision/version",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getCatalogVersion
});
