// @wix/auto_sdk_identity_verification/build/es/meta.mjs
// src/iam-verification-v1-start-response-verification.http.ts
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTBytesToSDKBytes } from "@wix/sdk-runtime/transformations/bytes";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixIamVerificationV1VerificationServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/_api/iam/verification",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/iam/verification",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/iam/verification",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_identity_verification";
function start(payload) {
  function __start({ host }) {
    const metadata = {
      entityFqdn: "wix.iam.verification.v1.start_response",
      method: "POST",
      methodFqn: "wix.iam.verification.v1.VerificationService.Start",
      packageName: PACKAGE_NAME,
      url: resolveWixIamVerificationV1VerificationServiceUrl({
        protoPath: "/v1/Start",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __start;
}
function verifyDuringAuthentication(payload) {
  function __verifyDuringAuthentication({ host }) {
    const metadata = {
      entityFqdn: "wix.iam.verification.v1.start_response",
      method: "POST",
      methodFqn: "wix.iam.verification.v1.VerificationService.VerifyDuringAuthentication",
      packageName: PACKAGE_NAME,
      url: resolveWixIamVerificationV1VerificationServiceUrl({
        protoPath: "/v1/auth/verify",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTBytesToSDKBytes,
          paths: [
            {
              path: "mfaChallengeData.verificationChallengeData.webauthnData.challenge"
            },
            {
              path: "mfaChallengeData.verificationChallengeData.webauthnData.allowCredentials.id"
            },
            {
              path: "mfaChallengeData.verificationChallengeData.webauthnData.extensions.largeBlobWrite"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "identity.createdDate" },
            { path: "identity.updatedDate" },
            { path: "identity.identityProfile.customFields.value.dateValue" },
            { path: "additionalData.*.dateValue" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "identity.identityProfile.customFields.value.numValue" },
            { path: "additionalData.*.numValue" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __verifyDuringAuthentication;
}
function resendDuringAuthentication(payload) {
  function __resendDuringAuthentication({ host }) {
    const metadata = {
      entityFqdn: "wix.iam.verification.v1.start_response",
      method: "POST",
      methodFqn: "wix.iam.verification.v1.VerificationService.ResendDuringAuthentication",
      packageName: PACKAGE_NAME,
      url: resolveWixIamVerificationV1VerificationServiceUrl({
        protoPath: "/v1/auth/resend",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTBytesToSDKBytes,
          paths: [
            {
              path: "mfaChallengeData.verificationChallengeData.webauthnData.challenge"
            },
            {
              path: "mfaChallengeData.verificationChallengeData.webauthnData.allowCredentials.id"
            },
            {
              path: "mfaChallengeData.verificationChallengeData.webauthnData.extensions.largeBlobWrite"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "identity.createdDate" },
            { path: "identity.updatedDate" },
            { path: "identity.identityProfile.customFields.value.dateValue" },
            { path: "additionalData.*.dateValue" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "identity.identityProfile.customFields.value.numValue" },
            { path: "additionalData.*.numValue" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __resendDuringAuthentication;
}

// src/iam-verification-v1-start-response-verification.meta.ts
function start2() {
  const payload = {};
  const getRequestOptions = start(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/Start",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function verifyDuringAuthentication2() {
  const payload = {};
  const getRequestOptions = verifyDuringAuthentication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/auth/verify",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function resendDuringAuthentication2() {
  const payload = {};
  const getRequestOptions = resendDuringAuthentication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/auth/resend",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  resendDuringAuthentication2 as resendDuringAuthentication,
  start2 as start,
  verifyDuringAuthentication2 as verifyDuringAuthentication
};
