// @wix/auto_sdk_redirects_redirects/build/internal/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  AttachPagesResponseStatus: () => AttachPagesResponseStatus,
  CallbackType: () => CallbackType,
  LocationType: () => LocationType,
  MembersAccountSection: () => MembersAccountSection,
  Prompt: () => Prompt,
  Status: () => Status,
  WebhookIdentityType: () => WebhookIdentityType,
  createRedirectSession: () => createRedirectSession4,
  onRedirectSessionCreated: () => onRedirectSessionCreated2
});
module.exports = __toCommonJS(index_exports);

// src/headless-v1-redirect-session-redirects.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/headless-v1-redirect-session-redirects.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/headless-v1-redirect-session-redirects.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixHeadlessV1RedirectSessionServiceUrl(opts) {
  const domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/_api/redirects-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/redirects-api",
        destPath: ""
      },
      {
        srcPath: "/redirect-session",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_redirects_redirects";
function createRedirectSession(payload) {
  function __createRedirectSession({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.redirect_session",
      method: "POST",
      methodFqn: "wix.headless.v1.RedirectSessionService.CreateRedirectSession",
      packageName: PACKAGE_NAME,
      url: resolveWixHeadlessV1RedirectSessionServiceUrl({
        protoPath: "/v1/redirect-session",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createRedirectSession;
}

// src/headless-v1-redirect-session-redirects.universal.ts
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["UNDEFINED"] = "UNDEFINED";
  LocationType2["OWNER_BUSINESS"] = "OWNER_BUSINESS";
  LocationType2["OWNER_CUSTOM"] = "OWNER_CUSTOM";
  LocationType2["CUSTOM"] = "CUSTOM";
  return LocationType2;
})(LocationType || {});
var Prompt = /* @__PURE__ */ ((Prompt2) => {
  Prompt2["login"] = "login";
  Prompt2["none"] = "none";
  return Prompt2;
})(Prompt || {});
var MembersAccountSection = /* @__PURE__ */ ((MembersAccountSection2) => {
  MembersAccountSection2["ACCOUNT_INFO"] = "ACCOUNT_INFO";
  MembersAccountSection2["BOOKINGS"] = "BOOKINGS";
  MembersAccountSection2["ORDERS"] = "ORDERS";
  MembersAccountSection2["SUBSCRIPTIONS"] = "SUBSCRIPTIONS";
  MembersAccountSection2["EVENTS"] = "EVENTS";
  return MembersAccountSection2;
})(MembersAccountSection || {});
var AttachPagesResponseStatus = /* @__PURE__ */ ((AttachPagesResponseStatus2) => {
  AttachPagesResponseStatus2["UNKNOWN"] = "UNKNOWN";
  AttachPagesResponseStatus2["SUCCESS"] = "SUCCESS";
  AttachPagesResponseStatus2["NO_ACTION"] = "NO_ACTION";
  AttachPagesResponseStatus2["ERROR"] = "ERROR";
  return AttachPagesResponseStatus2;
})(AttachPagesResponseStatus || {});
var CallbackType = /* @__PURE__ */ ((CallbackType2) => {
  CallbackType2["UNKNOWN"] = "UNKNOWN";
  CallbackType2["LOGOUT"] = "LOGOUT";
  CallbackType2["CHECKOUT"] = "CHECKOUT";
  CallbackType2["AUTHORIZE"] = "AUTHORIZE";
  return CallbackType2;
})(CallbackType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN"] = "UNKNOWN";
  Status2["SUCCESS"] = "SUCCESS";
  Status2["ERROR"] = "ERROR";
  return Status2;
})(Status || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createRedirectSession2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    bookingsCheckout: options?.bookingsCheckout,
    ecomCheckout: options?.ecomCheckout,
    eventsCheckout: options?.eventsCheckout,
    paidPlansCheckout: options?.paidPlansCheckout,
    login: options?.login,
    logout: options?.logout,
    auth: options?.auth,
    storesProduct: options?.storesProduct,
    bookingsBook: options?.bookingsBook,
    callbacks: options?.callbacks,
    preferences: options?.preferences,
    origin: options?.origin
  });
  const reqOpts = createRedirectSession(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          bookingsCheckout: "$[0].bookingsCheckout",
          ecomCheckout: "$[0].ecomCheckout",
          eventsCheckout: "$[0].eventsCheckout",
          paidPlansCheckout: "$[0].paidPlansCheckout",
          login: "$[0].login",
          logout: "$[0].logout",
          auth: "$[0].auth",
          storesProduct: "$[0].storesProduct",
          bookingsBook: "$[0].bookingsBook",
          callbacks: "$[0].callbacks",
          preferences: "$[0].preferences",
          origin: "$[0].origin"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/headless-v1-redirect-session-redirects.public.ts
function createRedirectSession3(httpClient) {
  return (options) => createRedirectSession2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onRedirectSessionCreated = (0, import_sdk_types.EventDefinition)(
  "wix.headless.v1.redirect_session_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths.transformPaths)(event, [
      {
        transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();

// src/headless-v1-redirect-session-redirects.context.ts
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createRedirectSession4 = /* @__PURE__ */ (0, import_rest_modules2.createRESTModule)(createRedirectSession3);
var onRedirectSessionCreated2 = (0, import_event_definition_modules.createEventModule)(onRedirectSessionCreated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AttachPagesResponseStatus,
  CallbackType,
  LocationType,
  MembersAccountSection,
  Prompt,
  Status,
  WebhookIdentityType,
  createRedirectSession,
  onRedirectSessionCreated
});
