// @wix/auto_sdk_redirects_redirects/build/internal/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createRedirectSession: () => createRedirectSession2
});
module.exports = __toCommonJS(meta_exports);

// src/headless-v1-redirect-session-redirects.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixHeadlessV1RedirectSessionServiceUrl(opts) {
  const domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/_api/redirects-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/redirects-api",
        destPath: ""
      },
      {
        srcPath: "/redirect-session",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_redirects_redirects";
function createRedirectSession(payload) {
  function __createRedirectSession({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.redirect_session",
      method: "POST",
      methodFqn: "wix.headless.v1.RedirectSessionService.CreateRedirectSession",
      packageName: PACKAGE_NAME,
      url: resolveWixHeadlessV1RedirectSessionServiceUrl({
        protoPath: "/v1/redirect-session",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createRedirectSession;
}

// src/headless-v1-redirect-session-redirects.meta.ts
function createRedirectSession2() {
  const payload = {};
  const getRequestOptions = createRedirectSession(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/redirect-session",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createRedirectSession
});
