// @wix/auto_sdk_stores_customizations-v-3/build/internal/es/meta.mjs
// src/stores-catalog-v3-customization-customizations-v-3.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/customization",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/stores/v3/customizations",
        destPath: "/v3/customizations"
      },
      {
        srcPath: "/stores/v3/bulk/customizations",
        destPath: "/v3/bulk/customizations"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/customization",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/customization",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/customization",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/customization",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/customization",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_customizations-v-3";
function createCustomization(payload) {
  function __createCustomization({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "customization.createdDate" },
          { path: "customization.updatedDate" },
          {
            path: "customization.choicesSettings.choices.image.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "customization.choicesSettings.choices.image.focalPoint.x" },
          { path: "customization.choicesSettings.choices.image.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "POST",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.CreateCustomization",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/customizations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "customization.createdDate" },
            { path: "customization.updatedDate" },
            {
              path: "customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCustomization;
}
function getCustomization(payload) {
  function __getCustomization({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "GET",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.GetCustomization",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/customizations/{customizationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "customization.createdDate" },
            { path: "customization.updatedDate" },
            {
              path: "customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCustomization;
}
function updateCustomization(payload) {
  function __updateCustomization({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "customization.createdDate" },
          { path: "customization.updatedDate" },
          {
            path: "customization.choicesSettings.choices.image.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "customization.choicesSettings.choices.image.focalPoint.x" },
          { path: "customization.choicesSettings.choices.image.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "PATCH",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.UpdateCustomization",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/customizations/{customization.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "customization.createdDate" },
            { path: "customization.updatedDate" },
            {
              path: "customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCustomization;
}
function deleteCustomization(payload) {
  function __deleteCustomization({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "DELETE",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.DeleteCustomization",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/customizations/{customizationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteCustomization;
}
function queryCustomizations(payload) {
  function __queryCustomizations({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "POST",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.QueryCustomizations",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/customizations/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "customizations.createdDate" },
            { path: "customizations.updatedDate" },
            {
              path: "customizations.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "customizations.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "customizations.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryCustomizations;
}
function bulkCreateCustomizations(payload) {
  function __bulkCreateCustomizations({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "customizations.createdDate" },
          { path: "customizations.updatedDate" },
          {
            path: "customizations.choicesSettings.choices.image.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "customizations.choicesSettings.choices.image.focalPoint.x" },
          { path: "customizations.choicesSettings.choices.image.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "POST",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.BulkCreateCustomizations",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/bulk/customizations/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.customization.createdDate" },
            { path: "results.customization.updatedDate" },
            {
              path: "results.customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "results.customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "results.customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateCustomizations;
}
function addCustomizationChoices(payload) {
  function __addCustomizationChoices({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "choices.image.urlExpirationDate" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "choices.image.focalPoint.x" },
          { path: "choices.image.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "POST",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.AddCustomizationChoices",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/customizations/{customizationId}/add-choices",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "customization.createdDate" },
            { path: "customization.updatedDate" },
            {
              path: "customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addCustomizationChoices;
}
function setCustomizationChoices(payload) {
  function __setCustomizationChoices({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "choices.image.urlExpirationDate" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "choices.image.focalPoint.x" },
          { path: "choices.image.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "POST",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.SetCustomizationChoices",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/customizations/{customizationId}/set-choices",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "customization.createdDate" },
            { path: "customization.updatedDate" },
            {
              path: "customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setCustomizationChoices;
}
function removeCustomizationChoices(payload) {
  function __removeCustomizationChoices({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "POST",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.RemoveCustomizationChoices",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/customizations/{customizationId}/remove-choices",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "customization.createdDate" },
            { path: "customization.updatedDate" },
            {
              path: "customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removeCustomizationChoices;
}
function bulkAddCustomizationChoices(payload) {
  function __bulkAddCustomizationChoices({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "customizationsChoices.choices.image.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "customizationsChoices.choices.image.focalPoint.x" },
          { path: "customizationsChoices.choices.image.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "POST",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.BulkAddCustomizationChoices",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/bulk/customizations/add-choices",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.customization.createdDate" },
            { path: "results.customization.updatedDate" },
            {
              path: "results.customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "results.customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "results.customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkAddCustomizationChoices;
}
function bulkUpdateCustomizations(payload) {
  function __bulkUpdateCustomizations({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "customizations.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "customizations.customization.createdDate" },
          { path: "customizations.customization.updatedDate" },
          {
            path: "customizations.customization.choicesSettings.choices.image.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "customizations.customization.choicesSettings.choices.image.focalPoint.x"
          },
          {
            path: "customizations.customization.choicesSettings.choices.image.focalPoint.y"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.customization",
      method: "POST",
      methodFqn: "wix.stores.catalog.customization.v3.CustomizationService.BulkUpdateCustomizations",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogCustomizationV3CustomizationServiceUrl({
        protoPath: "/v3/bulk/customizations/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.customization.createdDate" },
            { path: "results.customization.updatedDate" },
            {
              path: "results.customization.choicesSettings.choices.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "results.customization.choicesSettings.choices.image.focalPoint.x"
            },
            {
              path: "results.customization.choicesSettings.choices.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateCustomizations;
}

// src/stores-catalog-v3-customization-customizations-v-3.meta.ts
function createCustomization2() {
  const payload = {};
  const getRequestOptions = createCustomization(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/customizations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCustomization2() {
  const payload = { customizationId: ":customizationId" };
  const getRequestOptions = getCustomization(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/customizations/{customizationId}",
    pathParams: { customizationId: "customizationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCustomization2() {
  const payload = { customization: { id: ":customizationId" } };
  const getRequestOptions = updateCustomization(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/customizations/{customization.id}",
    pathParams: { customizationId: "customizationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCustomization2() {
  const payload = { customizationId: ":customizationId" };
  const getRequestOptions = deleteCustomization(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/customizations/{customizationId}",
    pathParams: { customizationId: "customizationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCustomizations2() {
  const payload = {};
  const getRequestOptions = queryCustomizations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/customizations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateCustomizations2() {
  const payload = {};
  const getRequestOptions = bulkCreateCustomizations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/customizations/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addCustomizationChoices2() {
  const payload = { customizationId: ":customizationId" };
  const getRequestOptions = addCustomizationChoices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/customizations/{customizationId}/add-choices",
    pathParams: { customizationId: "customizationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setCustomizationChoices2() {
  const payload = { customizationId: ":customizationId" };
  const getRequestOptions = setCustomizationChoices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/customizations/{customizationId}/set-choices",
    pathParams: { customizationId: "customizationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeCustomizationChoices2() {
  const payload = { customizationId: ":customizationId" };
  const getRequestOptions = removeCustomizationChoices(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/customizations/{customizationId}/remove-choices",
    pathParams: { customizationId: "customizationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkAddCustomizationChoices2() {
  const payload = {};
  const getRequestOptions = bulkAddCustomizationChoices(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/customizations/add-choices",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateCustomizations2() {
  const payload = {};
  const getRequestOptions = bulkUpdateCustomizations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/customizations/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  addCustomizationChoices2 as addCustomizationChoices,
  bulkAddCustomizationChoices2 as bulkAddCustomizationChoices,
  bulkCreateCustomizations2 as bulkCreateCustomizations,
  bulkUpdateCustomizations2 as bulkUpdateCustomizations,
  createCustomization2 as createCustomization,
  deleteCustomization2 as deleteCustomization,
  getCustomization2 as getCustomization,
  queryCustomizations2 as queryCustomizations,
  removeCustomizationChoices2 as removeCustomizationChoices,
  setCustomizationChoices2 as setCustomizationChoices,
  updateCustomization2 as updateCustomization
};
