// @wix/auto_sdk_stores_abandoned-carts/build/es/meta.mjs
// src/stores-v1-abandoned-cart-abandoned-carts.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixCartAbandonmentApiV1AbandonedCartsUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_abandoned-carts";
function getAbandonedCart(payload) {
  function __getAbandonedCart({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.v1.abandoned_cart",
      method: "GET",
      methodFqn: "wix.cart.abandonment.api.v1.AbandonedCarts.GetAbandonedCart",
      packageName: PACKAGE_NAME,
      url: resolveWixCartAbandonmentApiV1AbandonedCartsUrl({
        protoPath: "/v1/abandonedCarts/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCart.abandonTime" },
            { path: "abandonedCart.activities.timestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAbandonedCart;
}
function queryAbandonedCarts(payload) {
  function __queryAbandonedCarts({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.v1.abandoned_cart",
      method: "POST",
      methodFqn: "wix.cart.abandonment.api.v1.AbandonedCarts.QueryAbandonedCarts",
      packageName: PACKAGE_NAME,
      url: resolveWixCartAbandonmentApiV1AbandonedCartsUrl({
        protoPath: "/v1/abandonedCarts/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCarts.abandonTime" },
            { path: "abandonedCarts.activities.timestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAbandonedCarts;
}

// src/stores-v1-abandoned-cart-abandoned-carts.meta.ts
function getAbandonedCart2() {
  const payload = { id: ":id" };
  const getRequestOptions = getAbandonedCart(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/abandonedCarts/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryAbandonedCarts2() {
  const payload = {};
  const getRequestOptions = queryAbandonedCarts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/abandonedCarts/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getAbandonedCart2 as getAbandonedCart,
  queryAbandonedCarts2 as queryAbandonedCarts
};
