// @wix/stores_app-extensions/build/es/index.mjs
var __defProp = Object.defineProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};

// dashboard.ts
var dashboard_exports = {};
__export(dashboard_exports, {
  pages: () => dashboard_page_builders_exports,
  plugins: () => dashboard_pages_plugin_props_exports
});

// src/extensions/dashboard-page-builders.ts
var dashboard_page_builders_exports = {};
__export(dashboard_page_builders_exports, {
  productsList: () => productsList
});
function productsList() {
  return { pageId: "c821a776-85a9-4340-ae94-e08876c99944" };
}

// src/extensions/dashboard-pages-plugin-props.ts
var dashboard_pages_plugin_props_exports = {};
__export(dashboard_pages_plugin_props_exports, {
  Inventory: () => a8855e_515a_49c1_894a_035731d2fd44_exports,
  Products: () => c821a776_85a9_4340_ae94_e08876c99944_exports
});

// src/extensions/dashboard-page-plugins/59a8855e-515a-49c1-894a-035731d2fd44.ts
var a8855e_515a_49c1_894a_035731d2fd44_exports = {};

// src/extensions/dashboard-page-plugins/c821a776-85a9-4340-ae94-e08876c99944.ts
var c821a776_85a9_4340_ae94_e08876c99944_exports = {};
export {
  dashboard_exports as dashboard
};
