// @wix/stores/build/internal/es/meta.mjs
// meta.ts
import * as wishlist from "@wix/auto_sdk_stores_wishlist/meta";
import * as collections from "@wix/auto_sdk_stores_collections/meta";
import * as products from "@wix/auto_sdk_stores_products/meta";
import * as brandsV3 from "@wix/auto_sdk_stores_brands-v-3/meta";
import * as customizationsV3 from "@wix/auto_sdk_stores_customizations-v-3/meta";
import * as infoSectionsV3 from "@wix/auto_sdk_stores_info-sections-v-3/meta";
import * as inventoryItemsV3 from "@wix/auto_sdk_stores_inventory-items-v-3/meta";
import * as productsV3 from "@wix/auto_sdk_stores_products-v-3/meta";
import * as catalogVersioning from "@wix/auto_sdk_stores_catalog-versioning/meta";
import * as readOnlyVariantsV3 from "@wix/auto_sdk_stores_read-only-variants-v-3/meta";
import * as ribbonsV3 from "@wix/auto_sdk_stores_ribbons-v-3/meta";
import * as storesLocationsV3 from "@wix/auto_sdk_stores_stores-locations-v-3/meta";
import * as abandonedCarts from "@wix/auto_sdk_stores_abandoned-carts/meta";
import * as subscriptionOptions from "@wix/auto_sdk_stores_subscription-options/meta";
import * as inventory from "@wix/auto_sdk_stores_inventory/meta";
export {
  abandonedCarts,
  brandsV3,
  catalogVersioning,
  collections,
  customizationsV3,
  infoSectionsV3,
  inventory,
  inventoryItemsV3,
  products,
  productsV3,
  readOnlyVariantsV3,
  ribbonsV3,
  storesLocationsV3,
  subscriptionOptions,
  wishlist
};
