// @wix/auto_sdk_stores_wishlist/build/internal/es/meta.mjs
// src/platform-v1-wishlist-wishlist.http.ts
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressWishlistWishlistUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "ecom._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: "/api"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/stores/v1/wishlists",
        destPath: "/v1/wishlists"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_wishlist";
function getWishlistById(payload) {
  function __getWishlistById({ host }) {
    const metadata = {
      entityFqdn: "wix.platform.v1.wishlist",
      method: "POST",
      methodFqn: "com.wixpress.wishlist.Wishlist.GetWishlistById",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressWishlistWishlistUrl({
        protoPath: "/v1/wishlists/get",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "wishlist.items.dateAdded" }]
        }
      ])
    };
    return metadata;
  }
  return __getWishlistById;
}

// src/platform-v1-wishlist-wishlist.meta.ts
function getWishlistById2() {
  const payload = {};
  const getRequestOptions = getWishlistById(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/wishlists/get",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getWishlistById2 as getWishlistById
};
