// @wix/auto_sdk_stores_catalog-versioning/build/internal/es/index.mjs
// src/stores-catalog-v3-provision-catalog-versioning.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/stores-catalog-v3-provision-catalog-versioning.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixStoresCatalogProvisionV1StoresProvisionServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      },
      {
        srcPath: "/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    _: [
      {
        srcPath: "/_api/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_catalog-versioning";
function getCatalogVersion(payload) {
  function __getCatalogVersion({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.provision",
      method: "GET",
      methodFqn: "wix.stores.catalog.provision.v1.StoresProvisionService.GetCatalogVersion",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogProvisionV1StoresProvisionServiceUrl({
        protoPath: "/v3/provision/version",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCatalogVersion;
}

// src/stores-catalog-v3-provision-catalog-versioning.universal.ts
var Version = /* @__PURE__ */ ((Version2) => {
  Version2["V1_CATALOG"] = "V1_CATALOG";
  Version2["V3_CATALOG"] = "V3_CATALOG";
  Version2["STORES_NOT_INSTALLED"] = "STORES_NOT_INSTALLED";
  return Version2;
})(Version || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getCatalogVersion2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = getCatalogVersion(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/stores-catalog-v3-provision-catalog-versioning.public.ts
function getCatalogVersion3(httpClient) {
  return () => getCatalogVersion2(
    // @ts-ignore
    { httpClient }
  );
}

// src/stores-catalog-v3-provision-catalog-versioning.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getCatalogVersion4 = /* @__PURE__ */ createRESTModule(getCatalogVersion3);
export {
  DeleteStatus,
  Namespace,
  SiteCreatedContext,
  State,
  Version,
  WebhookIdentityType,
  getCatalogVersion4 as getCatalogVersion
};
