// @wix/auto_sdk_stores_catalog-versioning/build/internal/es/meta.mjs
// src/stores-catalog-v3-provision-catalog-versioning.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixStoresCatalogProvisionV1StoresProvisionServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/provision/version",
        destPath: "/v3/provision/version"
      },
      {
        srcPath: "/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ],
    _: [
      {
        srcPath: "/_api/stores/v3/provision/version",
        destPath: "/v3/provision/version"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_catalog-versioning";
function getCatalogVersion(payload) {
  function __getCatalogVersion({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.provision",
      method: "GET",
      methodFqn: "wix.stores.catalog.provision.v1.StoresProvisionService.GetCatalogVersion",
      packageName: PACKAGE_NAME,
      url: resolveWixStoresCatalogProvisionV1StoresProvisionServiceUrl({
        protoPath: "/v3/provision/version",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCatalogVersion;
}

// src/stores-catalog-v3-provision-catalog-versioning.meta.ts
function getCatalogVersion2() {
  const payload = {};
  const getRequestOptions = getCatalogVersion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/provision/version",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getCatalogVersion2 as getCatalogVersion
};
