// @wix/auto_sdk_redirects_redirects/build/meta.mjs
// src/headless-v1-redirect-session-redirects.http.ts
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixHeadlessV1RedirectSessionServiceUrl(opts) {
  const domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/_api/redirects-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/redirects-api",
        destPath: ""
      },
      {
        srcPath: "/redirect-session",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_redirects_redirects";
function createRedirectSession(payload) {
  function __createRedirectSession({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.redirect_session",
      method: "POST",
      methodFqn: "wix.headless.v1.RedirectSessionService.CreateRedirectSession",
      packageName: PACKAGE_NAME,
      url: resolveWixHeadlessV1RedirectSessionServiceUrl({
        protoPath: "/v1/redirect-session",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createRedirectSession;
}

// src/headless-v1-redirect-session-redirects.meta.ts
function createRedirectSession2() {
  const payload = {};
  const getRequestOptions = createRedirectSession(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/redirect-session",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createRedirectSession2 as createRedirectSession
};
