// @wix/auto_sdk_stores_wishlist/build/es/index.mjs
// src/platform-v1-wishlist-wishlist.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/platform-v1-wishlist-wishlist.http.ts
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressWishlistWishlistUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "ecom._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: "/api"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/stores/v1/wishlists",
        destPath: "/v1/wishlists"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_wishlist";
function getWishlistById(payload) {
  function __getWishlistById({ host }) {
    const metadata = {
      entityFqdn: "wix.platform.v1.wishlist",
      method: "POST",
      methodFqn: "com.wixpress.wishlist.Wishlist.GetWishlistById",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressWishlistWishlistUrl({
        protoPath: "/v1/wishlists/get",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "wishlist.items.dateAdded" }]
        }
      ])
    };
    return metadata;
  }
  return __getWishlistById;
}

// src/platform-v1-wishlist-wishlist.universal.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getWishlistById2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    limit: options?.limit,
    offset: options?.offset,
    kind: options?.kind
  });
  const reqOpts = getWishlistById(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.wishlist;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          limit: "$[1].limit",
          offset: "$[1].offset",
          kind: "$[1].kind"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/platform-v1-wishlist-wishlist.public.ts
function getWishlistById3(httpClient) {
  return (_id, options) => getWishlistById2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/platform-v1-wishlist-wishlist.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getWishlistById4 = /* @__PURE__ */ createRESTModule(getWishlistById3);
export {
  WebhookIdentityType,
  getWishlistById4 as getWishlistById
};
